File: vectic_mini.py
import math
class Vectic:
  def __init__(A,x,y):A.x=x;A.y=y        
  def __add__(A,v):return Vectic(v.x+A.x,v.y+A.y)
  def __iadd__(A,v):A.x+=v.x;A.y+=v.y;return A
  def __sub__(A,v):return Vectic(A.x-v.x,A.y-v.y)
  def __isub__(A,v):A.x-=v.x;A.y-=v.y;return A
  def __mul__(A,s):return Vectic(A.x*s,A.y*s)
  def __imul__(A,s):A.x*=s;A.y*=s;return A
  def __repr__(A):return f"Vectic({A.x}, {A.y})"
  def __truediv__(A,s):
    if isinstance(s,Vectic):return Vectic(A.x/s.x,A.y/s.y)
    return Vectic(A.x/s,A.y/s)
  def __floordiv__(A,s):
    if isinstance(s,Vectic):return Vectic(A.x//s.x,A.y//s.y)
    return Vectic(A.x//s,A.y//s)
  def __floor__(A):return A//1
  def __len__(A):return A.norm()
  def __eq__(A,v):return A.x==v.x and A.y==v.y
  def dist(A,v):return math.sqrt(A.dist2(v))
  def dist2(A,v):return(A.x-v.x)**2+(A.y-v.y)**2
  def norm(A):return A.dist(Vectic.zero())
  def normalized(A):return A/A.norm()
  def rot(A,t):return Vectic(A.x*math.cos(t)-A.x*math.sin(t),A.y*math.sin(t)+A.y*math.cos(t))
  def zero():return Vectic(0,0)

File: vectic_mini.lua
Vectic={}
Vectic={
	new=function(x,y)return{x=x,y=y}end,
	add=function(v,v2)return Vectic.new(v.x+v2.x,v.y+v2.y)end,
	sub=function(v,v2)return Vectic.new(v.x-v2.x,v.y-v2.y)end,
	mul=function(v,s)return Vectic.new(v.x*s,v.y*s)end,
	repr=function(v) return "Vectic.new("..v.x..", "..{v.y}..")"end,
	div=function(v,s)
	 if type(s)=="number" then return Vectic.new(v.x/s,v.y/s) end
	 return Vectic.new(v.x/s.x,v.y/s.y)
	end,
	floordiv=function(v,s)
	 if type(s)=="number"then return Vectic.new(v.x//s,v.y//s)end
	 return Vectic.new(v.x//s.x,v.y//s.y)
	end,
	floor=function(v)return Vectic.floordiv(v,1)end,
	dist2=function(v,v2)return(v.x-v2.x)^2+(v.y-v2.y)^2 end,
	dist=function(v,v2)return math.sqrt(v.dist2(v2))end,
	norm=function(v)return Vectic.dist(v,Vectic.zero())end,
	len=Vectic.norm,
	eq=function(v,v2)return v.x==v2.x and v.y==v2.y end,
	normalize=function(v)return Vectic.div(v,Vectic.norm(v))end,
	rotate=function(v,t)return Vectic.new(v.x*math.cos(t)-v.x*math.sin(t),v.y*math.sin(t)+v.y*math.cos(t))end,
	copy=function(v)return Vectic.new(v.x,v.y)end,
	zero=function()return Vectic.new(0,0)end,
	xy=function(v)return v.x,v.y end
}